modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinTextRangeProvider.java

```diff
@@ -362,7 +375,7 @@ private String GetText(int maxLength) {
         String text = (String)getAttribute(TEXT);
         if (text == null) return null;
         validateRange(text);
-        int endOffset = maxLength != -1 ? Math.min(end, start + maxLength) : end;
+        int endOffset = getValidStringIndex(start, maxLength, end);
```

Comment:
If I presume the arguments are (start, length, maxLength), the last two arguments on this line only need to be swapped.  They seem to be correct on LL 394, 498.

Reply:
I think the comment in https://github.com/openjdk/jdk/pull/10704 referred to an external project, so it was deemed as unrelated.

but this method, I feel, is rather convoluted.  why do we need to catch exceptions (malloc!) when we should be able to implement the required functionality with a few `if`s.

So I really have two comments: a) the code looks way too complex and b) if an average developer (like me) cannot figure out what it does within 10 seconds, a good comment would help.

Reply:
> So I really have two comments: a) the code looks way too complex and b) if an average developer (like me) cannot figure out what it does within 10 seconds, a good comment would help.

Sure thing.

I applied the one "usual pattern" of Integer overflow at a1c135abf7. Maybe, you like it.